plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadata.java

```diff
@@ -370,8 +371,8 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
                 jdbcClient.getTableHandle(session, tableName)
                         .ifPresent(tableHandle -> columns.put(tableName, getTableMetadata(session, tableHandle).getColumns()));
             }
-            catch (TableNotFoundException e) {
-                // table disappeared during listing operation
+            catch (TableNotFoundException | AccessDeniedException e) {
+                // table disappeared during listing operation or user is not access it
```

Comment:
not propagating `AccessDeniedException` can lead to actual problem being masked from user and system administrator

Reply:
Sure. We are not going to hide this in regular flow. However this method is used for metadata listing, where we hide any failures like that.

Reply:
Can this method be used in `DESC table` flow as well?

Reply:
It is used in DESC as well. So DESC is not failing with access denied, but return an empty results. It is because of the fact that `DESC` is a syntax sugar over a query from `information_schema`.

Reply:
To me this is a behavioral regression and I think we should fix this ASAP (or not merge this PR).

The fact that DESC is syntax sugar is _not_ something user should be concerned with, or should need to understand.

Reply:
This PR does not affect DESC. Notice that it is working in the same way when `TableNotException` is thrown. It won't report an error. This is another story.

Reply:
The fact that DESC returns 0 for a non existent table is a pre existing bug -- do we have an issue for this?

As i understand, this PR introduces a behavioral change in DESC that it now returns 0 for access denied as well. am i reading things correctly?

Reply:
You are right. 

I tried to fix it here: https://github.com/trinodb/trino/pull/6958

The fact it is access denied is maybe not that much important to DESC. If connector fails to deliver columns but was able to list a table or deliver table handle, the issue is swallowed.